Foundation/NSRunLoop.swift

```diff
@@ -95,10 +95,6 @@ extension NSRunLoop {
             return false
         }
         let modeArg = mode._cfObject
-        if _CFRunLoopFinished(_cfRunLoop, modeArg) {
-            return false
-        }
-        
```

Comment:
```swift

dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)) {
    let rl = NSRunLoop.currentRunLoop()
    let mode = "foo mode"
    let now = NSDate()
    let res = rl.runMode(mode, beforeDate: now.dateByAddingTimeInterval(10.0))
    assert(res == false)
    let t = NSDate().timeIntervalSinceReferenceDate - now.timeIntervalSinceReferenceDate
    assert(t < 10.0)
}

NSRunLoop.currentRunLoop().runUntilDate(NSDate.distantFuture())
```

or even the case

```swift
dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)) {
    let rl = NSRunLoop.currentRunLoop()
    let mode = "foo mode"
    let now = NSDate()
    let port = NSPort()
    rl.addPort(port, forMode: mode)
    rl.removePort(port, forMode: mode)
    let res = rl.runMode(mode, beforeDate: now.dateByAddingTimeInterval(10.0))
    assert(res == false)
    let t = NSDate().timeIntervalSinceReferenceDate - now.timeIntervalSinceReferenceDate
    assert(t < 10.0)
}

NSRunLoop.currentRunLoop().runUntilDate(NSDate.distantFuture())
```

Neither of these would work if the run loop is not claimed as finished at this point.